### ISSUE TITLE
Duplicate ID selectors on visits template

### ISSUE THREAD

**Issue by amdrew** [AffiliateWP/AffiliateWP#2309]
On the `dashboard-tabs-visits.php` template file, there are 2 elements that share an ID of `affwp-affiliate-dashboard-visits`:

https://github.com/AffiliateWP/AffiliateWP/blob/master/templates/dashboard-tab-visits.php#L1
https://github.com/AffiliateWP/AffiliateWP/blob/master/templates/dashboard-tab-visits.php#L24

As per HTML spec, there should only be one element with a unique ID, and in addition it causes a slight issue with the pagination's fragment identifier (https://github.com/AffiliateWP/AffiliateWP/blob/master/templates/dashboard-tab-visits.php#L75) since it doesn't know which ID to jump to.

I vote we:

1. Remove the ID from the table. IDs should only be used sparingly, and where necessary, such as for the fragment identifier.

2. Add a new class to the table. It can be the same as the old ID.

3. Update `forms.css`, since that uses the table's ID selector for column widths etc.

**Comment by amdrew** [comment]
`dashboard-tab-referrals.php` has the same issue so will need to be changed also.

**Comment by amdrew** [comment]
PR: https://github.com/AffiliateWP/AffiliateWP/pull/2310

**Comment by pippinsplugins** [comment]
I don't think we should remove the IDs. If anyone has styled off of those, we will have just borked their CSS.

**Comment by amdrew** [comment]
@pippinsplugins Good point. 

Perhaps we can put together a strategy for removing the IDs within the next 2-3 releases to give any user that has styled the table(s) plenty of time to update. 

Sidenote: Thoughts on best communicating changes like this with users? Dev blog like EDD? This won't be the last time we need to make similar updates.

In any case, I don't think we should leave something as-is (and technically broken) because we think users may have styled the ID selector. 

For backwards compatibility, how about we:

1. Keep the old IDs
2. Keep the new classes in the PR
3. Keep the old and new CSS (it's very minimal)
4. Remove old CSS and IDs in 2-3 releases.

**Comment by pippinsplugins** [comment]
That approach sounds good to me.

I have been wondering if it would be beneficial to have a dev blog. It would allow us to separate beta posts from actual release posts, though perhaps that's not a positive.

Another possibility is to have a single doc on our docs site that keeps an up-to-date list of all BC-breaking changes we make.

**Comment by amdrew** [comment]
@pippinsplugins I like the idea of starting small (and simple) and creating a single doc for any breaking changes 👍 

In addition, if we do have any breaking changes we should list them as the very first item in the changelog (and bold them).
